front/lib/api/assistant/generation.ts

```diff
@@ -392,7 +387,13 @@ export async function renderConversationForModelMultiActions({
     const c = r.value;
     if (tokensUsed + c <= allowedTokenCount) {
       tokensUsed += c;
-      selected.unshift(messages[i]);
+      if (messages[i].role === "content_fragment") {
+        const previousMessage = selected[0];
```

Comment:
technically there's nothing that ensures this is really a user message right ?

Reply:
I will add a check 👍